### ISSUE TITLE
Create a function where we can "copy" all annotations from child codes and also assign them to the parent code

### ISSUE THREAD

**Issue by miahass** [edgeryders/annotator_store-gem#167]
Potentially like the "copy" function where you can duplicate a code + its annotations, but for all of the child annotations of a parent, so we can assign all the annotations of the child codes to the parent code. Any way of doing this would be excellent.

**Comment by tanius** [comment]
The whole idea of having a code hierarchy with parent-child relations is that a parent code is automatically assumed to be applied to all annotations created with any of its child codes. (That is also the usual concept of using trees in user interfaces. For example, a directory is always assumed to contain all files contained in any of its sub-directories.)

That means, explicitly assigning all annotations of child codes to the parent code would create redundancy. And redundancy should be avoided because it will create inconsistencies between the (synonymous) copies further down the line.

So if there is any part of Open Ethnographer that does not yet consider a parent code's annotations to include all annotations of its child codes, that's a bug and should be reported.

Apart from that, I can imagine one other reason why you'd want this feature: when deleting a child code, users might want to transfer the annotations to the parent code. However, this case is already covered by the "merge code" feature, which does exactly that (transfer the annotations to the target code, which can be the parent, and then delete the code to be merged).

**tl;dr:** Could you please advice on the use case for the proposed feature so we can figure out a feature that is useful and also fits the concept of a code hierarchy.

**Comment by miahass** [comment]
‘So if there is any part of Open Ethnographer that does not yet consider a parent code's annotations to include all annotations of its child codes, that's a bug and should be reported.‘

Yes, so currently this doesn’t happen (parent codes don’t automatically take the annotations of child codes). If this were the case this would fix this issue, so we can take that approach instead!!

**Comment by miahass** [comment]
For some reason I thought this was on purpose, so I didn’t request that function... but yes, auto assigning all child annotations to the parent is ideal.

**Comment by tanius** [comment]
> parent codes don’t automatically take the annotations of child codes

Not by direct assignment, no. It is done though by a calculation at the time of showing the annotations of a parent code, which is functionally equivalent (and better, because it avoids redundancy). For example, it is implemented here:

* When you go to [the codes list](https://edgeryders.eu/annotator/codes) and select "Annotations" for any code in the list, you'll see a top-bar filter "Also include sub-codes: yes | no".
* In [the codes list](https://edgeryders.eu/annotator/codes), there are two numbers for the count of annotations for this code, such as "6+15". The first are the annotations directly assigned with this code, the second are the annotations of descendant codes.

Where else do you want to see parent codes automatically assigned to their child annotations?

**Comment by miahass** [comment]
Ah ok! I missed this function. Does this also translate to co-occurrences in Graph Ryder? I had thought not, but this is what I’m trying to accomplish with these parent annotations taking on the children. E.g. if depression co-occurs with loss, and depression is a child of mental health, will mental health Co-occur with loss?

**Comment by miahass** [comment]
So, to clarify--- what we have in the backend in Open Ethnographer is perfect as is. Just need to make sure that Graph Ryder shows the co-occurrences of the parent code with the co-occurrences of the child codes as described above with the example of depression and loss. Thank you :)

**Comment by tanius** [comment]
> what we have in the backend in Open Ethnographer is perfect as is

Right. Getting the co-occurrences to be calculated based on the hierarchical information is the responsibility of software like Graphryder that uses the Open Ethnographer API. The API supplies the hierarchy information in a field "ancestry" ([example](https://edgeryders.eu/annotator/codes/416.json)).

So please open an issue against Graphryder if necessary. Closing this one now.
